### PR TITLE
Add Outcome and Consensus Data Decoding and Tests

### PR DESCRIPTION
**Pull request by lalexgap**
Adds some decode methods for `outcome` and `consensusData` and adds some tests for encoding/decoding.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
   "homepage": "https://github.com/statechannels/nitro-protocol/",
   "scripts": {
     "prepare": "yarn build",
+    "test:typescript": "yarn jest --testPathIgnorePatterns='test/contract-tests' --reporters='default'",
```
Comment (kerzhner) [inline package.json]: Is this supposed to be `npx jest` instead of `yarn jest`?

File: src/contract/consensus-data.ts [hunk 2]
```diff
@@ -9,7 +9,53 @@ export interface ConsensusData {
...
+  };
+}
+
+export function finalVote(consensusData: ConsensusData): ConsensusData {
+  if (consensusData.furtherVotesRequired !== 0) {
+    throw new Error(
+      `Expected furtherVotesRequired to be 1, received ${consensusData.furtherVotesRequired} instead.`,
+    );
+  }
+  // TODO: if it's the final vote we return zeroed out consensusData
```
Comment (kerzhner) [inline src/contract/consensus-data.ts]: Is this TODO still needed?

File: src/contract/consensus-data.ts [hunk 3]
```diff
@@ -9,7 +9,53 @@ export interface ConsensusData {
...
+  return {furtherVotesRequired: 0, proposedOutcome: []};
+}
+
+export const voting = {
```
Comment (kerzhner) [inline src/contract/consensus-data.ts]: If this constant is exported, does every function above need to be exported?

Comment (lalexgap) [reply src/contract/consensus-data.ts]: Nope, I hadn't decided on which way to export them so I ended up exporting everything.

File: src/contract/consensus-data.ts (hunk 2 above)
Comment (lalexgap) [reply src/contract/consensus-data.ts]: No, I think this should just be a comment clarifying what we're doing.

**Pull request merged**